core/src/packet_deserializer.rs

```diff
@@ -82,40 +115,30 @@ impl PacketDeserializer {
         &self,
         recv_timeout: Duration,
         packet_count_upperbound: usize,
-    ) -> Result<(Vec<PacketBatch>, Option<SigverifyTracerPacketStats>), RecvTimeoutError> {
+    ) -> Result<(usize, Vec<BankingPacketBatch>), RecvTimeoutError> {
         let start = Instant::now();
-        let (mut packet_batches, mut aggregated_tracer_packet_stats_option) =
-            self.packet_batch_receiver.recv_timeout(recv_timeout)?;
-
-        let mut num_packets_received: usize = packet_batches.iter().map(|batch| batch.len()).sum();
-        while let Ok((packet_batch, tracer_packet_stats_option)) =
-            self.packet_batch_receiver.try_recv()
-        {
+        let message = self.packet_batch_receiver.recv_timeout(recv_timeout)?;
+        let packet_batches = &message.0;
+        let mut num_packets_received = packet_batches
+            .iter()
+            .map(|batch| batch.len())
+            .sum::<usize>();
```

Comment:
We now panic on overflow instead of handling it. Seems like a separate change from the rest of the PR.

Reply:
yep. i'll create a pr for this. https://github.com/solana-labs/solana/pull/25172/files#r1044407292

Reply:
>  i'll create a pr for this.

just done: https://github.com/solana-labs/solana/pull/29715